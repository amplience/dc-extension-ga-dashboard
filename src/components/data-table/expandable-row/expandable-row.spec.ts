import { render, fireEvent, screen } from '@testing-library/svelte';
import ExpandableRow from './expandable-row.svelte';
import { tick } from 'svelte';

describe('ExpandableRow', () => {
  it('should render the DataTable ExpandableRow component', () => {
    const { container } = render(ExpandableRow, {});

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the DataTable ExpandableRow and expand the row on click', async () => {
    const { container } = render(ExpandableRow, {});

    const element = screen.getByTestId('row-content');

    await fireEvent.click(element);
    await tick();

    expect(container.firstChild).toMatchSnapshot();
  });
});
