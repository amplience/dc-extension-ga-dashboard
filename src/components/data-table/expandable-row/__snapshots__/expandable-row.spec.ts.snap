// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpandableRow should render the DataTable ExpandableRow and expand the row on click 1`] = `
<div>
  <div
    class="data-table-expandable-row expanded svelte-34uld6"
  >
    <div
      class="row-content svelte-34uld6"
      data-testid="row-content"
    >
      <div
        class="svelte-cy4end"
      >
        <div
          class=" svelte-zpbvvv"
          style="--icon-width: 24px; --icon-height: 24px;"
        >
          <object
            src="ic-contractable.svg/"
          />
        </div>
      </div>
       
    </div>
     
    <div
      class="expandable-content svelte-34uld6"
    />
  </div>
</div>
`;

exports[`ExpandableRow should render the DataTable ExpandableRow component 1`] = `
<div>
  <div
    class="data-table-expandable-row  svelte-34uld6"
  >
    <div
      class="row-content svelte-34uld6"
      data-testid="row-content"
    >
      <div
        class="svelte-cy4end"
      >
        <div
          class=" svelte-zpbvvv"
          style="--icon-width: 24px; --icon-height: 24px;"
        >
          <object
            src="ic-expandable.svg/"
          />
        </div>
      </div>
       
    </div>
     
    <div
      class="expandable-content svelte-34uld6"
    />
  </div>
</div>
`;
